src/main/java/uk/gov/ons/ctp/response/notify/config/NotifyConfiguration.java

```diff
@@ -24,7 +26,32 @@
    * @return the notificationClient
    */
   @Bean
-  public NotificationClient notificationClient() {
-    return new NotificationClient(apiKey);
+  public NotificationClientApi notificationClient() {
+    validate();
+    return new ConfigurationAwareNotificationClient(this);
+  }
+
+  /**
+   * Method to validate the gov.notify api key and template id.  RuntimeException thrown as there really is no point
+   * in this service continuing if the keys are bogus
+   * @throws RuntimeException
```

Comment:
Not sure I like the use of RuntimeException either, might be better to have a NotificationClientConfigurationException. Spring will bail anyway if the bean creation fails.

Reply:
Checked exceptions are frowned upon these days

Reply:
Too much time away from Java ;) 

Though didn't necessarily mean checked exception you could extend Runtime, I just like dedicated exceptions. 

It's failing a codacy rule mind: https://www.codacy.com/app/sdcplatform/rm-notify-gateway/pullRequest?prid=1134224

Reply:
Fair point - I'll subclass it :-)

Reply:
Done